src/GameOptions/ui/KeyBindingPage.tsx

```diff
@@ -0,0 +1,308 @@
+    onClose();
+  };
+
+  return (
+    <Modal open={open} onClose={onClose}>
+      <div style={{ textAlign: "center" }}>
+        <Typography style={{ padding: "10px 20px" }}>Press the key you would like to use</Typography>
+        <Typography
+          minHeight="100px"
+          display="flex"
+          alignItems="center"
+          justifyContent="center"
+          margin="10px 0"
+          border="1px solid"
+        >
+          {parseKeyCombinationToString(combination)}
+        </Typography>
+        <Typography style={{ margin: "15px 0" }}>
+          {conflicts.size === 0 ? "No conflicts detected" : `Conflicts: ${[...conflicts]}`}
+        </Typography>
+        <div style={{ margin: "10px 0" }}>
+          <Button style={{ minWidth: "100px" }} onClick={onClickClear}>
+            Clear
+          </Button>
+          <Button style={{ marginLeft: "10px", minWidth: "100px" }} onClick={onClickDefault}>
+            Default
+          </Button>
+        </div>
+        <div>
+          <Button style={{ minWidth: "100px" }} onClick={onClickOK}>
+            OK
+          </Button>
+          <Button style={{ marginLeft: "10px", minWidth: "100px" }} onClick={onClickCancel}>
+            Cancel
+          </Button>
+        </div>
+      </div>
+    </Modal>
+  );
+}
+
+export function KeyBindingPage(): React.ReactElement {
+  const [popupOpen, setPopupOpen] = useState(false);
+  const [keyBindingType, setKeyBindingType] = useState<KeyBindingType>(SimplePage.Options);
+  const [isPrimary, setIsPrimary] = useState(true);
+
+  const showModal = (keyBindingType: KeyBindingType, isPrimary: boolean) => {
+    setPopupOpen(true);
+    setKeyBindingType(keyBindingType);
+    setIsPrimary(isPrimary);
+  };
+
+  const onClickHowToUse = () => {
+    dialogBoxCreate(
+      <>
+        <Typography>
+          You can assign up to 2 key combinations per "action". If a key combination is assigned to many actions,
+          pressing that key combination will perform all those actions.
+        </Typography>
+        <br />
+        <Typography>
+          Some key combinations cannot be used. Your OS and browsers usually have some built-in key bindings that cannot
+          be overridden. For example, on Windows, Windows+R always opens the "Run" dialog.
+        </Typography>
+        <br />
+        <Typography>
+          When you set up key bindings, the list of conflicts may contain "Endgame content". It means that the key
+          combination is currently used for features that you have not unlocked.
+        </Typography>
+        <br />
+        <Typography>
+          On non-Apple keyboards, the "Windows" key (other names: win, start, super, meta, etc.) is shown as ⊞. On Apple
+          keyboards, the command key is shown as ⌘.
+        </Typography>
+        <br />
+        <Typography>
+          Do NOT use the right Alt key and the AltGr key, especially if you don't use the US keyboard layout. On many
```

Comment:
You can remove this part. This is not an issue when using key-based binding (as opposed to keycode binding). Or at least, they are no more an issue than any other key that is done through more than a single keystroke. (Compose-based characters, function keys on laptops with a Fn key, etc.)